docs/tutorials/custom.rst

```diff
@@ -6,58 +6,111 @@ While Flambé offers a large number of `Component` objects to use in experiments
 researchers will typically need to use their own code, or modify one of our current
 component object.
 
-Flambé offers a simply mechanism to inject custom code in configs. Specifically,
-your code should be wrapped in a pip installable. This is as simple as including
-a setup.py alongside your module. For example:
 
-Say you have the following directory structure for your project:
+Writing your custom code
+------------------------
+
+Flambé configurations support any python object that inherits from ``flambe.Component``.
+You can decide to inherit from one of our base classes such as ``flambe.nn.Module`` or ``flambe.dataset.Dataset``,
+or you can inherit from ``flambe.Component`` directly.
+
+A ``Component`` must simply implement a ``run`` method which returns a boolean indicating whether execution
+should continue or not (useful for multi-step components such as a ``Trainer``).
+
+Additionally, if you would like to run hyperparameter search on your custom
+component, you must implement the ``metric`` method which returns the current best metric.
```

Comment:
For the first time something is mentioned in a section we should link instead of just formatting, i.e. use `` :meth:`~flambe.compile.Component.metric` ``

Note that the `~` character means only the last item, in this case `metric`, will be shown